app/assets/javascripts/components/revisions/diff_viewer.jsx

```diff
@@ -37,6 +37,18 @@ const DiffViewer = createReactClass({
     };
   },
 
+  // When 'show' is clicked, this component may or may not already have
+  // users data (a list of usernames) in its props. If it does, then 'show' will
+  // fetch the MediaWiki user ids, which are used for coloration. Those can't be
+  // fetched until the usernames are available, so 'show' will fetch the usernames
+  // first in that case. In that case, componentDidUpdate fetches the
+  // user ids as soon as usernames are avaialable.
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
This looks fine, but did you test it out manually on the Articles tab of a course with edited articles, as well as the Students tab?

This is behavior that is pretty important, but I don't think it's tested in any of feature specs.

Reply:
Yes, I did check the `diff_viewer.jsx` manually (the popup that shows the differences in the revised version form the previous one). I'll submit its feature spec in a new PR.